### PR TITLE
add transport layer

### PR DESCRIPTION
**Pull request by DrIhor**
Task #7

### CODE REVIEW

File: pkg/auth/routes/handler.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	return &HandlerItemsServ{
+		router:   router,
+		services: &auth.MockAuthService{},
+		ctx:      ctx,
+	}
+}
+
+func (h *HandlerItemsServ) HandlerItems() {
+	h.router.HandleFunc("/login", h.GetJWTToken).Methods("GET")
+	h.router.HandleFunc("/get-cert/{version}", h.GetCertKey).Methods("GET")
```
Comment (alex-a-renoire) [inline pkg/auth/routes/handler.go:29]: better use "certificate" instead of get-cert

File: pkg/auth/routes/handler.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("token", token)
+	w.WriteHeader(http.StatusCreated)
```
Comment (alex-a-renoire) [inline pkg/auth/routes/handler.go:55]: status "created" is for entities, i guess not for tokens. I would rather return 200 OK

File: pkg/auth/routes/handler.go (hunk 1 above)
Comment (DrIhor) [reply pkg/auth/routes/handler.go:29]: Oleksii requirements:
Auth service
/ login { UserName: string Password  (not empty) } -> JWT token {UserName, KeyVersion}
/ get-cert/{version}

File: pkg/auth/routes/handler.go (hunk 2 above)
Comment (DrIhor) [reply pkg/auth/routes/handler.go:55]: I create new JWT token based on user information. It seems to me that this thing is better to consider as entity.

**Review by alex-a-renoire — APPROVED** [review]

File: pkg/auth/server.go [hunk 3]
```diff
@@ -1 +1,98 @@
...
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type Server struct {
+	ctx    context.Context
```
Comment (romanilchyshyn) [inline pkg/auth/server.go:19]: Why do we need global ctx?

File: pkg/auth/server.go [hunk 4]
```diff
@@ -1 +1,98 @@
...
+	// run server
+	go func() {
+		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Fatal(err)
+		}
+	}()
+
+	fmt.Println("Server is running on: " + s.getHTTPAddress())
+
+	<-ctx.Done() // wait end of work
```
Comment (romanilchyshyn) [inline pkg/auth/server.go:81]: Will it ever happen?

File: pkg/auth/server.go (hunk 3 above)
Comment (DrIhor) [reply pkg/auth/server.go:19]: To use graceful shutdown at whole server if such process exist(connections, DB etc)

Comment (romanilchyshyn) [reply pkg/auth/server.go:19]: Do we have such usage right now?

Comment (romanilchyshyn) [reply pkg/auth/server.go:19]: Because this add like 200 lines of dead code. Could be added separately if needed.

File: pkg/auth/server.go (hunk 4 above)
Comment (DrIhor) [reply pkg/auth/server.go:81]: In current situation waiting shut down from system(and will handle at cmd part)

**Review by romanilchyshyn — APPROVED** [review]

**Review by stasBigunenko — APPROVED** [review]

**Pull request merged**